lib/cassandra/executors.rb

```diff
@@ -80,6 +80,7 @@ def run
 
         loop do
           tasks = nil
+          @term ||= nil
```

Comment:
I think you should be able to get away with moving `@term = ...` above `@pool = ...` in the constructor and remove `@term ||= nil` let me know if it works